rpcs3/rpcs3qt/log_frame.cpp

```diff
@@ -506,11 +506,13 @@ void log_frame::UpdateUI()
 
 				// set scrollbar to max means auto-scroll
 				sb->setValue(is_max ? sb->maximum() : sb_pos);
+
+				// Limit processing time
+				if (steady_clock::now() >= start + 4ms) break;
 			}
 		}
 
-		// Limit processing time
-		if (steady_clock::now() >= start + 4ms || buf.empty()) break;
+		m_tty_size = new_size;
```

Comment:
this is still wrong though.
I just noticed that you don't use the actual pos, but the ideal new_size.
The position might be smaller than new_size, if the loop is postponed for GUI updates.

Reply:
Not sure what you mean by using pos, its irrelevant for GUI. (only exists for sys_tty_write to function properly)
Fixed some bug with that 4ms limit though.